lib/gltf/gltf-builder.ts

```diff
@@ -276,33 +276,89 @@ export class GLTFBuilder {
     box: Box3D,
     defaultMaterialIndex: number,
   ): Promise<void> {
-    // Separate mesh triangles by face orientation
-    const topTriangles: NonNullable<typeof box.mesh>["triangles"] = []
-    const bottomTriangles: NonNullable<typeof box.mesh>["triangles"] = []
+    // Only apply board bounds filtering for panels (multiple boards)
+    // For single boards, all top/bottom triangles get the texture
+    const hasMultipleBoards = box.boardBounds && box.boardBounds.length > 1
+
+    const isInsideAnyBoard = (circuitX: number, circuitY: number): boolean => {
+      if (!hasMultipleBoards) {
+        return true
+      }
+      for (const bounds of box.boardBounds!) {
+        if (
+          circuitX >= bounds.minX &&
+          circuitX <= bounds.maxX &&
+          circuitY >= bounds.minY &&
+          circuitY <= bounds.maxY
+        ) {
+          return true
+        }
+      }
+      return false
+    }
+
+    const isTriangleInsideBoard = (
+      triangle: NonNullable<typeof box.mesh>["triangles"][0],
+    ): boolean => {
+      if (!hasMultipleBoards) {
+        return true
+      }
+      const centerX =
+        (triangle.vertices[0].x +
+          triangle.vertices[1].x +
+          triangle.vertices[2].x) /
+        3
+      const centerZ =
+        (triangle.vertices[0].z +
+          triangle.vertices[1].z +
+          triangle.vertices[2].z) /
+        3
+
+      const circuitX = centerX + box.center.x
+      const circuitY = -centerZ + box.center.z
+
+      return isInsideAnyBoard(circuitX, circuitY)
+    }
```

Comment:
remove these closures, tons of top-level file pollution DONT POLLUTE TOP LEVEL FILES!!!